grr/server/grr_response_server/fleetspeak_utils.py

```diff
@@ -112,3 +113,34 @@ def GetLabelsFromFleetspeak(client_id):
       grr_labels.append(fs_label.label)
 
   return grr_labels
+
+
+def GetClientIdsFromFleetspeak() -> List[Text]:
+  """Returns client IDs of Fleetspeak-enabled database.
+
+  Returns:
+    A list of Client IDs.
+  """
+  res = fleetspeak_connector.CONN.outgoing.ListClients(admin_pb2.ListClientsRequest())
+  if not res.clients or not res.clients[0].client_id:
+    return []
+  clients_list = list(res.clients)
+  client_ids_list = [FleetspeakIDToGRRID(client.client_id) for client in clients_list]
+  return list(client_ids_list)
+
+
+def FetchClientResourceUsageRecordsFromFleetspeak(client_id: Text, limit: int) -> List[ClientResourceUsageRecord]:
+  """Returns aggregated resource usage metrics of a client in Fleetspeak-enabled database.
+
+  Args:
+    client_id: Id of the client to fetch Fleetspeak resource usage records for.
+    limit: Max number of resource usage records to retrieve.
+
+  Returns:
+    A list of client resource usage records retrieved from Fleetspeak.
+  """
+  res = fleetspeak_connector.CONN.outgoing._stub.FetchClientResourceUsageRecords(
```

Comment:
Why do you need to use `CONN.outgoing._stub` and not simply `CONN.outgoing`. `_stub`, given that its name starts with an underscore, is a semi-private attribute that is likely an implementation detail of the `CONN.outgoing` object.

Reply:
You're right, that's definitely a better way of achieving this goal. I implemented a new method in class `OutgoingConnection` in Fleetspeak to support it, you can check it out here google/fleetspeak@1947d45.